### PR TITLE
Fix flaky integration test for ProvisioningRequest

### LINKED ISSUE #2483: Flaky integration test for provisioning AdmissionCheck
**What happened**:

Test: `Provisioning when A workload is using a provision admission check Should not set AdmissionCheck status to Rejected, deactivate Workload, emit an event, and bump metrics when workloads is not Finished, and the ProvisioningRequest's condition is set to CapacityRevoked`

```
{Timed out after 5.000s.
The function passed to Eventually failed at /home/prow/go/src/kubernetes-sigs/kueue/test/integration/controller/admissionchecks/provisioning/provisioning_test.go:663 with:
Expected
    <v1beta1.CheckState>: Pending
to equal
    <v1beta1.CheckState>: Ready failed [FAILED] Timed out after 5.000s.
The function passed to Eventually failed at /home/prow/go/src/kubernetes-sigs/kueue/test/integration/controller/admissionchecks/provisioning/provisioning_test.go:663 with:
Expected
    <v1beta1.CheckState>: Pending
to equal
    <v1beta1.CheckState>: Ready
In [It] at: /home/prow/go/src/kubernetes-sigs/kueue/test/integration/controller/admissionchecks/provisioning/provisioning_test.go:669 @ 06/26/24 17:35:44.329
}
```

**What you expected to happen**:

No flakiness

**How to reproduce it (as minimally and precisely as possible)**:

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-kueue-test-integration-main/1806017708857233408

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #2483 

#### Special notes for your reviewer:
There was a race condition and the AdmissionCheck didn't get `Ready` state before finishing a Workload. 

Also fixed typos and test case description.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/controller/admissionchecks/provisioning/provisioning_test.go [hunk 1]
```diff
@@ -635,14 +634,23 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 
+			ginkgo.By("Checking if an AdmissionCheck is Ready", func() {
```
Comment (alculquicondor) [inline test/integration/controller/admissionchecks/provisioning/provisioning_test.go:637]: Is this enough? or should we also check that the Workload is Admitted?
I guess they might happen in the same API update.

Comment (PBundyra) [reply test/integration/controller/admissionchecks/provisioning/provisioning_test.go:637]: I don't think it matters whether it's Admitted or not in this scenario

**Pull request merged**